scripts/west_commands/runners/linkserver.py

```diff
@@ -0,0 +1,236 @@
+        # be able to find the shared library that tells us what
+        # version of the tools we're using.
+        self.linkserver = os.fspath(
+            Path(self.require(self.linkserver)).resolve())
+        self.logger.info(f'LinkServer Version: {self.linkserver_version_str}')
+
+        #rtos = self.thread_info_enabled and self.supports_thread_info
+
+        if command == 'flash':
+            self.flash(**kwargs)
+        else:
+            linkserver_cmd = ([self.linkserver] +
+                              ["gdbserver"]    +
+                              ["--probe", "#"+str(self.probe) ] +
+                              ["--gdb-port", str(self.gdb_port )] +
+                              ["--semihost-port", str(self.semihost_port) ] +
+			      self.override_cli +
+                              [self.device])
+
+            if command in ('debug', 'attach'):
+            # self.print_gdbserver_message()
+                if self.elf_name is  None or not os.path.isfile(self.elf_name):
+                    raise ValueError('Cannot debug; elf file required')
+
+                gdb_cmd = ([self.gdb_cmd] +
+                           self.tui_arg +
+                          [self.elf_name] +[
+                          '-ex', 'target remote {}:{}'.format(self.gdb_host, self.gdb_port)])
+
+                if command == 'debug':
+                    gdb_cmd += [ '-ex', 'load', '-ex', 'monitor reset']
+
+                if command == 'attach':
+                    linkserver_cmd += ['--attach']
+
+                self.run_server_and_client(linkserver_cmd, gdb_cmd)
+
+            elif command == 'debugserver':
+                if self.gdb_host:
+                    raise ValueError('Cannot run debugserver with --gdb-host')
+
+                self.check_call(linkserver_cmd)
+
+            return
+
+    def do_erase(self, **kwargs):
+
+        linkserver_cmd = ([self.linkserver, "flash"] + ["--probe", "#"+str(self.probe)] +
+                          [self.device] + ["erase"])
+        self.logger.debug("flash erase command = " + str(linkserver_cmd))
+        self.check_call(linkserver_cmd)
+
+    def _build_override_cli(self):
+
+        override_cli = []
+
+        if self.override is not None:
+            for ov in self.override:
+                override_cli = (override_cli + [ "-o", str(ov)])
+
+        return override_cli
+
+    def flash(self, **kwargs):
+
+        linkserver_cmd = ([self.linkserver, "flash"] + ["--probe", "#"+str(self.probe)] + self.override_cli + [self.device])
+
+        if self.erase:
+            self.do_erase()
+
+        if self.bin_name is not None and os.path.isfile(self.bin_name):
+            if self.dt_flash:
+                load_addr = self.flash_address_from_build_conf(self.build_conf)
+            else:
+                self.logger.critical("no load flash address could be found...")
+                raise RuntimeError("missing argument")
```

Comment:
What's going on here? It seems like you always want to flash using partition info from devicetree/.config. The error doesn't seem related to a missing argument.

Reply:
I believe all of your comments have been addressed.